VSCode extension does not work with dev builds of the language server

```
[DEBUG] Extension activated.
[DEBUG] Python extension is available but not yet active, activating...
[DEBUG] Using python from the python extension: python
[DEBUG] Python version 3.9.4
[DEBUG] Current version: 0.6.1.dev167
[DEBUG] Last update was 2021-05-08T17:27:47.122Z
[ERROR] TypeError: Invalid Version: 0.6.1.dev167
```

Fixing this would help test potential fixes

--- a/code/changes/168.fix.rst
+++ b/code/changes/168.fix.rst
@@ -1,0 +1,1 @@
+It's now possible to test dev builds of the language server with the extension

--- a/code/src/language-server.ts
+++ b/code/src/language-server.ts
@@ -151,6 +151,14 @@ export class LanguageServerBootstrap {
       let version = await getInstalledVersion(this.python)
       this.logger.debug(`Current version: ${version}`)
 
+      // PEP 440 dev release numbers (https://www.python.org/dev/peps/pep-0440/#developmental-releases)
+      // are not compatible with semver. To allow people to test dev builds with the extension we
+      // need to transform the version number to be semver compatible.
+      if (version.includes(".dev")) {
+        version = version.replace(".dev", "-dev.")
+        this.logger.debug(`Semver compatible version: ${version}`)
+      }
+
       return version
     } catch (err) {
       this.logger.debug(`${err.message}`)

